zk/datadog_checks/zk/zk.py

```diff
@@ -260,11 +262,16 @@ def parse_stat(self, buf):
         # >= 3.4.4.
         start_line = buf.readline()
         match = self.version_pattern.search(start_line)
+        # this is to grab the additional version information
+        total_match = self.metadata_version_pattern.search(start_line)
         if match is None:
             return (None, None, "inactive", None)
             raise Exception("Could not parse version from stat command output: %s" % start_line)
         else:
             version = match.group()
+            # grabs the entire version number.
+            metadata_version = total_match.group(1)
+            self.set_metadata('version', metadata_version)
```

Comment:
I think this version parsing section can be simplified. For example, you can first grab `3.4.10-39d3a4f269333c922ed3db283be479f9deacaa0f` part for `metadata_version`, then split by `-` to keep only `3.4.10` for the initial `version`.

Reply:
I made the changes in this commit: https://github.com/DataDog/integrations-core/pull/4927/commits/20dbbbf3a50ccee6bd97a08ddc5a619664912b1c